src/realm/db.cpp

```diff
@@ -1666,6 +1666,19 @@ void DB::close_internal(std::unique_lock<InterprocessMutex> lock, bool allow_ope
     }
 }
 
+bool DB::waiting_for_write_lock() const
+{
+    SharedInfo* info = m_file_map.get_addr();
+
+    uint32_t next_ticket = info->next_ticket.load(std::memory_order_relaxed);
+    uint32_t next_served = info->next_served.load(std::memory_order_relaxed);
+
+    // There is no thread waiting for the write lock if either:
+    //  a) No thread currently holds the write lock, or
+    //  b) One thread holds the lock and no other thread attempted to acquire it.
+    return !(next_ticket == next_served || next_ticket == next_served + 1);
+}
```

Comment:
There is a problem here: if designed for use by some thread A already holding the write lock, and the intent is to return true if some OTHER thread is waiting... then that other thread will have incremented next_ticket, so you will merely need:

* someone_waiting = next_ticket > next_served

Reply:
I believe the condition is not enough. It should be `next_ticket > next_served + 1`. If a thread holds the write mutex and there is no other thread waiting, then: next_ticket = my_ticket + 1 and next_served = my_ticket.

Reply:
@danieltabacaru is correct :-)